### PR TITLE
isUpToDate flag on channel state

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "stream-chat",
-  "version": "2.9.0",
+  "version": "2.9.1-dev.0",
```
Comment (mahboubii) [inline package.json:3]:
```suggestion
  "version": "2.9.0",
```

File: src/channel_state.ts [hunk 2]
```diff
@@ -424,6 +437,10 @@ export class ChannelState<
     }
   }
 
+  setIsUptoDate = (isUpToDate: boolean) => {
```
Comment (mahboubii) [inline src/channel_state.ts:440]: does it make sense to move this to the channel level?

Comment (mahboubii) [inline src/channel_state.ts:440]: some jsdoc would also be good explaining this similar to this.isUpToDate

File: src/channel.ts [hunk 3]
```diff
@@ -1547,13 +1547,19 @@ export class Channel<
         if (event.message) {
           /* if message belongs to current user, always assume timestamp is changed to filter it out and add again to avoid duplication */
           const ownMessage = event.user?.id === this.getClient().user?.id;
-          s.addMessageSorted(event.message, ownMessage);
+
+          if (this.state.isUpToDate || event.message.parent_id) {
```
Comment (mahboubii) [inline src/channel.ts:1551]: thread message with `show_in_channel` flag? maybe disable this completely for threaded messages

**Review by mahboubii — COMMENTED** [review]
could you also add a few tests for it

File: src/channel_state.ts (hunk 2 above)
Comment (vishalnarkhede) [reply src/channel_state.ts:440]: Its only about messages, so I thought state would be a better place!!

File: src/channel.ts (hunk 3 above)
Comment (vishalnarkhede) [reply src/channel.ts:1551]: Good catch

**Review by mahboubii — APPROVED** [review]
@vishalnarkhede looks good 👍, could you also add some tests for it?

**Pull request merged**